sbt-twirl/src/main/scala/play/twirl/sbt/TemplateProblem.scala

```diff
@@ -87,7 +88,7 @@ object TemplateProblem {
     }
 
     def apply(source: Option[File], codec: Codec): TemplateMapping = {
-      val lines = source.toSeq flatMap { file => IO.readLines(file, codec.charSet) }
+      val lines = source.toSeq flatMap { file => TwirlIO.readFileAsString(file, codec).split('\n') }
```

Comment:
`TemplateMapping` also assumes a new line is one character, so the position in the file won't line up with the line start anymore.

Actually I think it would be better for `TemplateMapping`'s constructor to accept the whole file string. The fact that it accepts lines and assumes one-character newlines seems like bad API. That change would introduce some binary compatibility issues so we'd probably want to do it separately, though.